br/pkg/lightning/backend/local/duplicate.go

```diff
@@ -460,39 +593,22 @@ func (manager *DuplicateManager) getValuesFromRegion(
 		return errors.Errorf("key error")
 	}
 
-	maxKeyLen := 0
-	for _, kv := range resp.Pairs {
-		l := manager.keyAdapter.EncodedLen(kv.Key)
-		if l > maxKeyLen {
-			maxKeyLen = l
-		}
-	}
-	buf := make([]byte, maxKeyLen)
-
 	log.L().Error("get keys", zap.Int("key size", len(resp.Pairs)))
-	for i := 0; i < maxRetryTimes; i++ {
-		b := manager.db.NewBatch()
-		opts := &pebble.WriteOptions{Sync: false}
-		for _, kv := range resp.Pairs {
-			encodedKey := manager.keyAdapter.Encode(buf, kv.Key, 0, 0)
-			b.Set(encodedKey, kv.Value, opts)
-			if b.Count() > maxWriteBatchCount {
-				err = b.Commit(opts)
-				if err != nil {
-					break
-				} else {
-					b.Reset()
-				}
-			}
-		}
-		if err == nil {
-			err = b.Commit(opts)
-		}
-		if err == nil {
-			return nil
-		}
+
+	rawRows := make([][]byte, 0, len(resp.Pairs))
+	for i, kv := range resp.Pairs {
+		rawRows = append(rawRows, kv.Value)
+		handles.dataConflictInfos[i].Row = decoder.DecodeRawRowDataAsStr(handles.handles[i], kv.Value)
```

Comment:
I think we shall check whether `kv.Key` equals `handles[i].Key`.  I'm not sure TiKV will always return keys in order which we send keys to it.